### PR TITLE
Add isSentinelValue

### PR DESCRIPTION
**Pull request by skShekhar**
This PR adds a predicate to determine if a value is a sentinel value. #793

### CODE REVIEW

**Comment by skShekhar** [comment]
@char0n Can you review this PR

**Comment by char0n** [comment]
@skShekhar thanks! I'll provide a review ASAP.

File: src/isSentinelValue.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * Checks whether the passed value is a sentinel value.
+ *
+ * @func isSentinelValue
+ * @memberOf RA
+ * @since {@link https://char0n.github.io/ramda-adjunct/2.33.0|v2.33.0}
+ * @category Type
+ * @sig * -> Boolean
+ * @param {*} val The value to test
+ * @return {boolean}
+ * @see {@link RA.isSentinelValue|isSentinelValue}
```
Comment (char0n) [inline src/isSentinelValue.js:15]:
```suggestion
 * @see {@link RA.isSentinelValue|isSentinelValue}
```
No need to mention the reference to function to itself.

File: src/isSentinelValue.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * @example
+ *
+ * RA.isSentinelValue(-1); //=> true
+ *
+ * RA.isSentinelValue('-1'); //=> false
+ * RA.isSentinelValue(1); //=> false
+ * RA.isSentinelValue([-1]); //=> false
+ */
+const isSentinelValue = curryN(1, (val) => isInteger32(val) && val === -1);
```
Comment (char0n) [inline src/isSentinelValue.js:24]: Can we change this to idiomatic sentinel value check? The `isInteger` predicate already guarantees that value is signed 32 bit integer before it moves to `~val === 0`. If eslint is giving you issues, use `eslint-disable-line no-bitwise`.

```suggestion
const isSentinelValue = curryN(1, (val) => isInteger32(val) && ~val === 0);
```

File: src/isSentinelValue.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { curryN } from 'ramda';
+
+import isInteger32 from './isInteger32';
+
+/**
+ * Checks whether the passed value is a sentinel value.
```
Comment (char0n) [inline src/isSentinelValue.js:6]:
```suggestion
 * Checks whether the passed value is {@link https://github.com/getify/You-Dont-Know-JS/blob/9959fc904d584bbf0b02cf41c192f74ff4238581/types-grammar/ch4.md#the-curious-case-of-the-|a sentinel value}.
```

File: types/index.d.ts [hunk 4]
```diff
@@ -455,6 +455,11 @@ declare namespace RamdaAdjunct {
          */
         isSparseArray(val: any): boolean;
 
+        /**
+         * Checks whether the passed value is a sentinel value.
```
Comment (char0n) [inline types/index.d.ts:459]:
```suggestion
         * Checks whether the passed value is {@link https://github.com/getify/You-Dont-Know-JS/blob/9959fc904d584bbf0b02cf41c192f74ff4238581/types-grammar/ch4.md#the-curious-case-of-the-|a sentinel value}.

**Review by char0n — CHANGES_REQUESTED** [review]
Great job! Can you please look into my CR suggestion before we move forward?

File: src/isSentinelValue.js (hunk 2 above)
Comment (skShekhar) [reply src/isSentinelValue.js:24]: Is there any reason why you prefer `~val === 0` over `val === -1`. Isn't sentinel value directly defined as -1?

Comment (char0n) [reply src/isSentinelValue.js:24]: Technically both are equivalent. The reason I prefer former (`~val === 0`) is that this operation actually performs a complement of the value and then we assert if the complement equals `+0` or `-0` and `0` respectively. This actually describe what the algorithm for computing sentinel value actually is rather then just asserting for `-1`.

I guess that's my reasoning about it.

File: src/isSentinelValue.js (hunk 1 above)
Comment (skShekhar) [reply src/isSentinelValue.js]: Removed the reference

File: src/isSentinelValue.js (hunk 2 above)
Comment (skShekhar) [reply src/isSentinelValue.js]: Addressed

File: src/isSentinelValue.js (hunk 3 above)
Comment (skShekhar) [reply src/isSentinelValue.js]: Addressed

File: types/index.d.ts (hunk 4 above)
Comment (skShekhar) [reply types/index.d.ts]: Addressed

**Review by char0n — APPROVED** [review]

**Pull request merged**